platforms/allwinner-d1/d1-config/src/lib.rs

```diff
@@ -0,0 +1,75 @@
+#![no_std]
+
+use serde::{Serialize, Deserialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PlatformConfig {
+    pub i2c: I2cConfiguration,
+    pub i2c_puppet: I2cPuppetConfiguration,
+    pub blink_service: LedBlinkService,
+}
+
+// I2C
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct I2cConfiguration {
+    #[serde(default)]
+    pub enabled: bool,
+    #[serde(default = "I2cConfiguration::default_mapping")]
+    pub mapping: Mapping,
+}
+
+impl I2cConfiguration {
+    const fn default_mapping() -> Mapping {
+        Mapping::LicheeRvTwi0
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum Mapping {
+    LicheeRvTwi0,
```

Comment:
this is backwards; it's TWI2 on the lichee rv and TWI0 on the mq pro.

Reply:
Good catch! Will fix.